helpers/ajustesHelper/ajustes.movimientos.helper.go

```diff
@@ -0,0 +1,287 @@
+			return err
+		}
+	}
+
+	for _, nv := range novedades {
+		if _, err := movimientosArkaHelper.PutNovedadElemento(nv, nv.Id); err != nil {
+			return err
+		}
+	}
+
+	return nil
+
+}
+
+// separarElementosPorSalida Separa los elementos según el tipo de ajuste de cada uno y los agrupa según la salida. Además retorna los elementos actualizados
+func separarElementosPorSalida(elementos []*models.ElementosMovimiento, updateVls, updateSg, updateMp []*models.DetalleElemento_) (elementosSalidas map[int]*models.ElementosPorActualizarSalida, pendientes_ []*models.DetalleElemento_, actualizados []*models.ElementosMovimiento, outputError map[string]interface{}) {
+
+	funcion := "separarElementosPorSalida"
+	defer errorctrl.ErrorControlFunction(funcion+" - Unhandled Error!", "500")
+
+	elementosSalidas = make(map[int]*models.ElementosPorActualizarSalida)
+	for _, el := range elementos {
+
+		if len(updateMp) > 0 {
+			if idx := findElementoInArrayD(updateMp, el.ElementoActaId); idx > -1 {
+				if updateMp[idx].ValorResidual == el.ValorResidual && updateMp[idx].VidaUtil == el.VidaUtil {
+					updateMp = append(updateMp[:idx], updateMp[idx+1:]...)
+				} else {
+					elemento_ := creaNuevoElementoMovimiento(updateMp[idx], el)
+					actualizados = append(actualizados, elemento_)
+				}
+				continue
+			}
+		} else if el.MovimientoId.EstadoMovimientoId.Nombre != "Salida Aprobada" {
+			continue
+		}
+
+		if len(updateSg) > 0 {
+			if idx := findElementoInArrayD(updateSg, el.ElementoActaId); idx > -1 {
+				if elementosSalidas[el.MovimientoId.Id] == nil {
+					elementosSalidas[el.MovimientoId.Id] = new(models.ElementosPorActualizarSalida)
+					elementosSalidas[el.MovimientoId.Id].Salida = el.MovimientoId
+				}
+				elemento_ := creaNuevoElementoMovimiento(updateSg[idx], el)
+				actualizados = append(actualizados, elemento_)
+
+				elementosSalidas[el.MovimientoId.Id].UpdateSg = append(elementosSalidas[el.MovimientoId.Id].UpdateSg, updateSg[idx])
+			}
+		} else if len(updateVls) > 0 {
+			if idx := findElementoInArrayD(updateVls, el.ElementoActaId); idx > -1 {
+				if elementosSalidas[el.MovimientoId.Id] == nil {
+					elementosSalidas[el.MovimientoId.Id] = new(models.ElementosPorActualizarSalida)
+					elementosSalidas[el.MovimientoId.Id].Salida = el.MovimientoId
+				}
+				elemento_ := creaNuevoElementoMovimiento(updateVls[idx], el)
+				actualizados = append(actualizados, elemento_)
+
+				elementosSalidas[el.MovimientoId.Id].UpdateVls = append(elementosSalidas[el.MovimientoId.Id].UpdateVls, updateVls[idx])
+			}
+		}
+
+	}
+	return elementosSalidas, updateMp, actualizados, nil
+
+}
+
+// generarMovimientoAjuste Crea el registro del movimiento de inventario y contable resultantes del ajuste
+func generarMovimientoAjuste(sg, vls, msc, mp []*models.DetalleElemento_, movContables []*models.MovimientoTransaccion) (movimiento *models.Movimiento, trContable *models.TransaccionMovimientos, outputError map[string]interface{}) {
```

Comment:
Partir al menos así:
```suggestion
func generarMovimientoAjuste(sg, vls, msc, mp []*models.DetalleElemento_,
movContables []*models.MovimientoTransaccion) (movimiento *models.Movimiento,
trContable *models.TransaccionMovimientos, outputError map[string]interface{}) {
```
De aplicarse via web este _suggested change_, abrir luego en local este archivo y guardarlo para ver si se agregan/quitan espacios